File: nbc-product-store/Utilities/ValidationUtility.cs
using nbc_product_store.Constants;
using nbc_product_store.Models.Cart;
using nbc_product_store.Models.Error;

namespace nbc_product_store.Utilities
{
    public static class ValidationUtility
    {
        public static ServiceError ValidateCartRequest(CartRequest request)
        {
            ServiceError se = null;

            if (request == null || request.ProductId <= 0 || request.Quantity <= 0)
            {
                return new ServiceError(AppConstants.INVALID_REQUEST_STATUS_CODE, AppConstants.INVALID_REQUEST_STATUS_DESCRIPTION);
            }

            return se;
        }

    }
}

File: java-product-store/src/main/java/com/nbc/productstore/utils/ValidationUtility.java
package com.nbc.productstore.utils;

import com.nbc.productstore.models.CartRequest;
import com.nbc.productstore.models.ServiceError;

public class ValidationUtility {
    public static ServiceError validateCartRequest(CartRequest request) {
        if (request == null) return new ServiceError("INVALID_REQUEST", "Request is null");
        if (request.getProductId() <= 0) return new ServiceError("INVALID_PRODUCT_ID", "productId must be > 0");
        if (request.getQuantity() <= 0) return new ServiceError("INVALID_QUANTITY", "quantity must be > 0");
        return null;
    }
}
